src/Engines/TypesenseEngine.php

```diff
@@ -2,116 +2,222 @@
 
 namespace Typesense\LaravelTypesense\Engines;
 
+use Typesense\LaravelTypesense\Typesense;
 use Exception;
-use Laravel\Scout\Builder;
-use Laravel\Scout\Engines\Engine;
-use Illuminate\Support\Collection;
-use Illuminate\Support\LazyCollection;
 use Illuminate\Database\Eloquent\Model;
-use Typesense\LaravelTypesense\Typesense;
 use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Support\Collection;
+use Illuminate\Support\LazyCollection;
+use Illuminate\Support\Str;
+use Laravel\Scout\Builder;
+use Laravel\Scout\Engines\Engine;
 
 /**
- * Class TypesenseEngine
+ * Class TypesenseSearchEngine.
  *
- * @package Typesense\LaravelTypesense\Engines
  * @date    4/5/20
+ *
  * @author  Abdullah Al-Faqeir <abdullah@devloops.net>
  */
 class TypesenseEngine extends Engine
 {
-
+    /**
+     * @var \Typesense\LaravelTypesense\Typesense
+     */
     private Typesense $typesense;
 
     /**
-     * typesenseEngine constructor.
+     * @var array
+     */
+    private array $groupBy = [];
+
+    /**
+     * @var int
+     */
+    private int $groupByLimit = 3;
+
+    /**
+     * @var string
+     */
+    private string $startTag = '<mark>';
+
+    /**
+     * @var string
+     */
+    private string $endTag = '</mark>';
+
+    /**
+     * @var int
+     */
+    private int $limitHits = -1;
+
+    /**
+     * @var array
+     */
+    private array $locationOrderBy = [];
+
+    /**
+     * TypesenseSearchEngine constructor.
      *
-     * @param  \Typesense\LaravelTypesense\Typesense  $typesense
+     * @param \Typesense\LaravelTypesense\Typesense $typesense
      */
     public function __construct(Typesense $typesense)
     {
         $this->typesense = $typesense;
     }
 
     /**
-     * @param  \Illuminate\Database\Eloquent\Collection  $models
+     * @param \Illuminate\Database\Eloquent\Collection<int, Model>|Model[] $models
      *
      * @throws \Http\Client\Exception
      * @throws \JsonException
      * @throws \Typesense\Exceptions\TypesenseClientError
+     * @noinspection NotOptimalIfConditionsInspection
      */
     public function update($models): void
     {
         $collection = $this->typesense->getCollectionIndex($models->first());
 
-        if (config('scout.soft_delete', false) && $this->usesSoftDelete($models->first())) {
+        if ($this->usesSoftDelete($models->first()) && $models->first()->softDelete) {
```

Comment:
Any specific reason for stopping checking the config value, which is a scout feature?